### PR TITLE
chore: introduce sonar cloud validation

### PR DESCRIPTION
**Pull request by manolo**
This is a draft on top of @Artur- 's branch bringing java 11 to master

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -731,6 +742,18 @@
                 <excludedGroups></excludedGroups>
             </properties>
         </profile>
+        <profile>
+            <id>sonar-cloud</id>
+            <properties>
+                <sonar.java.source>11</sonar.java.source>
```
Comment (Artur-) [inline pom.xml:748]: Should this be `${maven.compiler.source}`?

File: .github/workflows/sonar.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |
+          mvn -B verify -DskipTests -pl \!flow-plugins/flow-gradle-plugin \
```
Comment (Artur-) [inline .github/workflows/sonar.yml:36]: Why do we skip gradle?

File: pom.xml [hunk 3]
```diff
@@ -68,17 +68,17 @@
             https://maven.vaadin.com/vaadin-addons
         </flow.addons.repo.url>
 
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
```
Comment (Artur-) [inline pom.xml:71]: Is there any actual need to change this at the same time or can we just enable Sonar cloud first?

Comment (manolo) [reply pom.xml:71]: no, just wanted to test it with jdk 11

File: pom.xml (hunk 1 above)
Comment (manolo) [reply pom.xml]: reverted

File: .github/workflows/sonar.yml (hunk 2 above)
Comment (manolo) [reply .github/workflows/sonar.yml]: added it back

**Review by Artur- — APPROVED** [review]

**Pull request merged**